pkg/infrastructure/azure/azure.go

```diff
@@ -201,49 +201,55 @@ func (p *Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput
 	storageURL := fmt.Sprintf("https://%s.blob.%s", storageAccountName, session.Environment.StorageEndpointSuffix)
 	blobURL := fmt.Sprintf("%s/%s/%s", storageURL, containerName, blobName)
 
-	// Create storage account
-	createStorageAccountOutput, err := CreateStorageAccount(ctx, &CreateStorageAccountInput{
-		SubscriptionID:     subscriptionID,
-		ResourceGroupName:  resourceGroupName,
-		StorageAccountName: storageAccountName,
-		CloudName:          platform.CloudName,
-		Region:             platform.Region,
-		AuthType:           session.AuthType,
-		Tags:               tags,
-		CustomerManagedKey: platform.CustomerManagedKey,
-		TokenCredential:    tokenCredential,
-		ClientOpts:         p.clientOptions,
-	})
-	if err != nil {
-		return err
-	}
-
-	storageAccount := createStorageAccountOutput.StorageAccount
-	storageClientFactory := createStorageAccountOutput.StorageClientFactory
-	storageAccountKeys := createStorageAccountOutput.StorageAccountKeys
-
-	logrus.Debugf("StorageAccount.ID=%s", *storageAccount.ID)
+	var storageAccount *armstorage.Account
+	var storageClientFactory *armstorage.ClientFactory
+	var storageAccountKeys []armstorage.AccountKey
+
+	var createStorageAccountOutput *CreateStorageAccountOutput
+	if platform.CloudName != azure.StackCloud {
+		// Create storage account
+		createStorageAccountOutput, err = CreateStorageAccount(ctx, &CreateStorageAccountInput{
+			SubscriptionID:     subscriptionID,
+			ResourceGroupName:  resourceGroupName,
+			StorageAccountName: storageAccountName,
+			CloudName:          platform.CloudName,
+			Region:             platform.Region,
+			AuthType:           session.AuthType,
+			Tags:               tags,
+			CustomerManagedKey: platform.CustomerManagedKey,
+			TokenCredential:    tokenCredential,
+			ClientOpts:         p.clientOptions,
+		})
+		if err != nil {
+			return err
+		}
+		storageAccount = createStorageAccountOutput.StorageAccount
+		storageClientFactory = createStorageAccountOutput.StorageClientFactory
+		storageAccountKeys = createStorageAccountOutput.StorageAccountKeys
 
-	// Create blob storage container
-	publicAccess := armstorage.PublicAccessNone
-	createBlobContainerOutput, err := CreateBlobContainer(ctx, &CreateBlobContainerInput{
-		SubscriptionID:       subscriptionID,
-		ResourceGroupName:    resourceGroupName,
-		StorageAccountName:   storageAccountName,
-		ContainerName:        containerName,
-		PublicAccess:         to.Ptr(publicAccess),
-		StorageClientFactory: storageClientFactory,
-	})
-	if err != nil {
-		return err
+		logrus.Debugf("StorageAccount.ID=%s", *storageAccount.ID)
 	}
 
-	blobContainer := createBlobContainerOutput.BlobContainer
-	logrus.Debugf("BlobContainer.ID=%s", *blobContainer.ID)
-
 	// Upload the image to the container
 	_, skipImageUpload := os.LookupEnv("OPENSHIFT_INSTALL_SKIP_IMAGE_UPLOAD")
 	if !(skipImageUpload || platform.CloudName == azure.StackCloud) {
+		// Create vhd blob storage container
```

Comment:
question: If I understand it right, this has a slight change in existing behaviour. Previously, blob container is always created regardless of skipping image or not. 

With this change, blob container is only created if image upload is not skip (assuming non azure stack platform). This is OK, right?

Reply:
Yes you got it right. AzureStack expects the VHD (this is the raw source file for RHCOS which is used to create the os image) to be uploaded in advance. So users would create their own storage container.

Reply:
Oh cool, thanks!